### ISSUE TITLE
Upgrading from 1.10 to 1.14 causes memory exhaustion.

### ISSUE THREAD

**Issue by hero-david** [newrelic/newrelic-fluent-bit-output#122]
**Bug description**

NewRelic logging pods shoot up to consume as much memory as their specified resource limit allows, then restart.

**Version of Helm and Kubernetes**

Kubernetes:
```console
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.5", GitCommit:"34f89fd3fb1a106e1b23d3454b2f2cbf305602a1", GitTreeState:"clean", BuildDate:"2023-02-10T14:48:59Z", GoVersion:"go1.19.4", Compiler:"gc", Platform:"linux/amd64"}
```

Helm:
```console
version.BuildInfo{Version:"v3.10.3", GitCommit:"835b7334cfe2e5e27870ab3ed4135f136eecc704", GitTreeState:"clean", GoVersion:"go1.18.9"}
```

**Which chart?**

nri-bundle 5.0.4

**What happened?**

We upgraded from the NewRelic bundle chart version 3.4.0 to 5.0.4. Upon doing so I saw that the newrelic-logging pods were restarting often, and on further inspection, it looks like they are using much more memory than they were before, causing OOMKilled errors.

The default memory limit is 128Mi, but we had previously upped that value to 256Mi.
I updated the limit again to 512Mi, but the same pattern of memory climb, then restart occurs.

With regards to the specific forwarder image, this brought the newrelic-fluentbit-output image from version 1.10.0 to 1.14.0:
![image](https://user-images.githubusercontent.com/108767410/224847601-c66b60e2-376d-438e-aaec-69b9f019cc08.png)

I have since rolled back the image version to 1.10, and we are seeing much lower memory usage, without the flapping / OOMKill cycle that we experienced with 1.14. This is potentially an issue with the upstream fluentbit version, but I have looked into the underlying lib version for each of the above output containers.

**What you expected to happen?**

I would have expected the same memory usage patterns to occur despite an upgrade, especially given that the [default memory limits have not changed from 128Mi](https://github.com/newrelic/helm-charts/blob/master/charts/newrelic-logging/values.yaml#L183)

**How to reproduce it?**

1. Install the nri bundle @ 3.0.4 with the values below. 
2. Observe the memory usage of newrelic-logging on a cluster with at least 10 pods per node generating logs.
3. Upgrade the nri-bundle to @ 5.0.4 with the values below.
4. Observe the memory usage on the same cluster / node compared to the previous 1.10 version

 
Values are as supplied

```yaml
global:
  # licenseKey:
  # cluster: 
  lowDataMode: true
  tolerations:
    - ...

...

newrelic-logging:
  # newrelic-logging.enabled -- Install the [`newrelic-logging` chart](https://github.com/newrelic/helm-charts/tree/master/charts/newrelic-logging)
  enabled: true
  fluentBit:
    criEnabled: true
  resources:
    limits:
      memory: 256Mi # Default [128Mi](https://github.com/newrelic/helm-charts/tree/master/charts/newrelic-logging#resources)

```

**Comment by hero-david** [comment]
It looks like the fluentbit versions are as such:
v1.10.0: [1.8.1](https://github.com/newrelic/newrelic-fluent-bit-output/blob/v1.11.0/Dockerfile#L21)
v1.14.0: [1.9.4](https://github.com/newrelic/newrelic-fluent-bit-output/blob/v1.14.0/Dockerfile#L21)

**Comment by jsubirat** [comment]
Hey @hero-david , our latest v1.14.2 image is now based on Fluent Bit 1.9.9. Could you please check if you're still experiencing this issue with this newer version? Thanks!

**Comment by jsubirat** [comment]
Hi @hero-david , we have upgraded the Fluent Bit version in our Docker images, now the v1.16.0 images are based on Fluent Bit 2.0.8. This version should fix your memory problems. You can find this image in the [newrelic-logging:1.14.1 Helm chart](https://github.com/newrelic/helm-charts/blob/master/charts/newrelic-logging/Chart.yaml#LL4C1-L4C16), which is now available as part of the {{nri-bundle:5.0.15}} Helm chart as well. I'll proceed to close the issue, don't hesitate to open another if you find any other issue. Thanks a lot!

**Comment by hero-david** [comment]
Hi @jsubirat , apologies for the delay in response, this upgrade fell a little in priority after the blocker.
We've just updated to the 1.16.0 image, everything looks much better (Without the memory utilization thrashing), and resource usage pre and post upgrade is quite comparable, which is what we expected to see.

Thank you for keeping this issue updated, it has been much appreciated!
